app/services/user_flash_updater.rb

```diff
@@ -15,7 +15,7 @@ def set_flash_message
   private
 
   def needs_to_confirm_phone_change?
-    form.phone_changed?
+    form.require_phone_confirmation?
```

Comment:
This is related to my previous comment about this not being necessary since `require_phone_confirmation?` is equal to `phone_changed?`